shell/platform/common/BUILD.gn

```diff
@@ -0,0 +1,21 @@
+# Copyright 2014 The Chromium Authors. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+source_set("common") {
+  sources = [
+    "dart_platform_init.cc",
+  ]
+
+  defines = []
+  if (is_debug) {
+    defines += ["DEBUG"]
+  } else {
+    defines += ["NDEBUG"]
+  }
```

Comment:
This shouldn't be needed.  NDEBUG should already be set properly.  If DEBUG is needed by Dart VM and isn't being set, then we should be able to teach the gn targets inside the dart repo to push that define onto us.